src/renderer_gl.cpp

```diff
@@ -6992,6 +6992,42 @@ namespace bgfx { namespace gl
 					, bx::uint32_imax(depth, 1)
 					) );
 			}
+			else
+			{
+				BX_ASSERT(depth <= 1, "Blitting 3D regions is not supported");
+
+				GLuint fbo;
+				GL_CHECK(glGenFramebuffers(1, &fbo) );
+
+				GL_CHECK(glBindFramebuffer(GL_FRAMEBUFFER, fbo) );
+
+				GL_CHECK(glFramebufferTexture2D(GL_FRAMEBUFFER
+						, GL_COLOR_ATTACHMENT0
+						, GL_TEXTURE_2D
+						, src.m_id
+						, bi.m_srcMip
+						) );
+
+				BX_ASSERT(GL_FRAMEBUFFER_COMPLETE == glCheckFramebufferStatus(GL_FRAMEBUFFER)
+					, "glCheckFramebufferStatus failed 0x%08x"
+					, glCheckFramebufferStatus(GL_FRAMEBUFFER)
+				);
+
+				GL_CHECK(glActiveTexture(GL_TEXTURE0) );
+				GL_CHECK(glBindTexture(GL_TEXTURE_2D, dst.m_id) );
+
+				GL_CHECK(glCopyTexSubImage2D(GL_TEXTURE_2D
+					, bi.m_dstMip
+					, bi.m_dstX
+					, bi.m_dstY
+					, bi.m_srcX
+					, bi.m_srcY
+					, width
+					, height) );
+
+				GL_CHECK(glBindFramebuffer(GL_FRAMEBUFFER, 0) );
+				GL_CHECK(glDeleteFramebuffers(1, &fbo) );
```

Comment:
This seems like huge performance issue. Create/destroy FBO per blit.

Reply:
Yes, this is probably costly (the fast path is there for a reason).
I could cache the FBO itself, but the binding and attachment must to be done each time and its where most of the validation cost occurs.
Am I missing a way to do it better ?

Reply:
Maybe something like this: for each src texture, cache with the FBO already validated, so at least we don't pay the cost of revalidation for repeatedly blitting from the same src texture. 
The FBO handle could be stored in the TextureGL struct, so we can invalidate it when the texture is changed.

Reply:
See the second commit for a proposal of this idea. Ideally the fbos should be cached by (texture, mip), but it may bloat TextureGL too much to put an array of FBO of sufficient size for all levels. Here I just store the last mip used for blit. Let me know what you think